x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/search_and_filter_bar.tsx

```diff
@@ -106,7 +108,7 @@ export const SearchAndFilterBar: React.FunctionComponent<SearchAndFilterBarProps
                 showAgentActivityTour={showAgentActivityTour}
               />
             </EuiFlexItem>
-            {!cloud?.isServerlessEnabled && (
+            {authz.fleet.allAgents && authz.fleet.allSettings && !cloud?.isServerlessEnabled ? (
```

Comment:
Should we simplify and introduce new authz property 
`authz.fleet.addAgents : authz.fleet.allAgents && authz.fleet.readSettings`
`authz.fleet.addFleetServer: authz.fleet.allAgents && authz.fleet.allSettings`